pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -389,10 +396,26 @@ func (e *PluginManager) Finalize(ctx context.Context, tCtx pluginsCore.TaskExecu
 		// the same event (idempotent) and then come here again...
 		err = e.ClearFinalizers(ctx, o)
 		if err != nil {
-			return err
+			errs.Append(err)
 		}
 	}
-	return nil
+
+	// If we should delete the resource when finalize is called, do a best effort delete.
+	if config.GetK8sPluginConfig().DeleteResourceOnFinalize {
+		// Attempt to delete resource, if not found, return success.
+		if err := e.kubeClient.GetClient().Delete(ctx, o); err != nil {
+			if IsK8sObjectNotExists(err) {
+				return errs.ErrorOrDefault()
+			}
+
+			// This happens sometimes because a node gets removed and K8s deletes the pod. This will result in a
+			// Pod does not exist error. This should be retried using the retry policy
```

Comment:
> This should be retried using the retry policy
is this descriptive or meant to be a todo?

Reply:
Should we make this a plugin property or just blanket?

Reply:
I guess we can offer an override per plugin? I was debating doing that or whether that was an overkill.. but yeah we can add that...

Reply:
Added [here](https://github.com/flyteorg/flyteplugins/pull/184/files)